component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -277,50 +283,78 @@ private String getErrorPage(AuthenticationContext context) throws Authentication
-            // Adds the claims to the profile if the user enrol and continued.
-            Map<String, String> claims = new HashMap<>();
-            if (context.getProperty(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL) != null) {
-                claims.put(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL,
-                        context.getProperty(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL).toString());
-            }
-            if (context.getProperty(TOTPAuthenticatorConstants.QR_CODE_CLAIM_URL) != null) {
-                claims.put(TOTPAuthenticatorConstants.QR_CODE_CLAIM_URL,
-                        context.getProperty(TOTPAuthenticatorConstants.QR_CODE_CLAIM_URL).toString());
-            }
-            try {
-                TOTPKeyGenerator.addTOTPClaimsAndRetrievingQRCodeURL(claims, username, context);
-            } catch (TOTPException e) {
-                throw new AuthenticationFailedException("Error while adding TOTP claims to the user : " + username, e);
-            }
-        }
-        if (token != null) {
-            try {
-                int tokenValue = Integer.parseInt(token);
-                if (!isValidTokenLocalUser(tokenValue, username, context)) {
-                    throw new AuthenticationFailedException("Authentication failed, user :  " + username);
-                }
-                if (StringUtils.isNotBlank(username)) {
-                    AuthenticatedUser authenticatedUser = new AuthenticatedUser();
-                    authenticatedUser.setAuthenticatedSubjectIdentifier(username);
-                    authenticatedUser.setUserName(
-                            UserCoreUtil.removeDomainFromName(MultitenantUtils.getTenantAwareUsername(username)));
-                    authenticatedUser.setUserStoreDomain(UserCoreUtil.extractDomainFromName(username));
-                    authenticatedUser.setTenantDomain(MultitenantUtils.getTenantDomain(username));
-                    context.setSubject(authenticatedUser);
-                } else {
-                    context.setSubject(AuthenticatedUser.createLocalAuthenticatedUserFromSubjectIdentifier(username));
-                }
-            } catch (TOTPException | NumberFormatException e) {
-                throw new AuthenticationFailedException("TOTP Authentication process failed for user " + username, e);
-            }
-        }
-    }
+												 AuthenticationContext context) throws AuthenticationFailedException {
+
+		String token = request.getParameter(TOTPAuthenticatorConstants.TOKEN);
+		String username = context.getProperty("username").toString();
+		AuthenticatedUser authenticatedUserObject =
+				(AuthenticatedUser) context.getProperty(TOTPAuthenticatorConstants.AUTHENTICATED_USER);
+		String tenantDomain = MultitenantUtils.getTenantDomain(username);
+		String userStoreDomain = UserCoreUtil.extractDomainFromName(username);
+		boolean isLocalUser = TOTPUtil.isLocalUser(context);
+		if (isLocalUser &&
```

Comment:
It's better if u can move the validation to a separate method

Reply:
Fixed in 5d35b84